compiler/rustc_hir_analysis/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -129,6 +129,18 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         if self.missing_lifetimes() { "lifetime" } else { "generic" }
     }
 
+    fn is_trait(&self) -> bool {
+        self.tcx.is_trait(self.def_id)
+    }
+
+    fn is_in_trait_impl(&self) -> bool {
```

Comment:
I fear that this will affect the wrong-number-of-generics suggestions when they show up in the self type of a trait impl.

I think we should just check that the path segment associated with this error is contained in this trait ref: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/struct.Impl.html#structfield.of_trait

Reply:
Thanks. Done.